### PR TITLE
Feature/create open

### PR DESCRIPTION
**Pull request by nkp123**
Good luck & have fun

### CODE REVIEW

File: daemon/src/communication/ServerListener.cpp [hunk 1]
```diff
@@ -120,12 +120,14 @@ void* executor(void* connector)
...
-			static_cast<OperationWithPathRequest*>(request)->getMode());
+			static_cast<OperationWithPathRequest*>(request)->getMode(),
+			static_cast<OperationWithPathRequest*>(request)->getPid());
 		break;
 	case OperationWithPathRequest::Open:
 		response = diskOps->open(
 			static_cast<OperationWithPathRequest*>(request)->getPath(),
-			static_cast<OperationWithPathRequest*>(request)->getMode());
+			static_cast<OperationWithPathRequest*>(request)->getMode(),
+			static_cast<OperationWithPathRequest*>(request)->getPid());
```
Comment (lychanl) [inline daemon/src/communication/ServerListener.cpp]: PID is already in conn (see getting FD few lines above)
please remove it from request

File: daemon/src/management/FileDescriptor.h [hunk 2]
```diff
@@ -40,6 +41,11 @@ struct FileDescriptor
 {
     const static unsigned char M_READ = 1;
     const static unsigned char M_WRITE = 2;
+    const static unsigned char M_CREATE = 4;
+    const static unsigned char M_CREATE_PERM_R = 8;
+    const static unsigned char M_CREATE_PERM_W = 16;
+    const static unsigned char M_CREATE_PERM_X = 32;
+
```
Comment (lychanl) [inline daemon/src/management/FileDescriptor.h]: I assume those are fully compatibile with flags like O_CREATE, O_WRONLY etc.?

File: utils/src/OperationWithPathRequest.cpp [hunk 3]
```diff
@@ -48,6 +48,16 @@ void OperationWithPathRequest::serialize(char* data)
...
+int OperationWithPathRequest::getPid()
+{
+	return pid;
+}
+
+void OperationWithPathRequest::setPid(int pid)
+{
+	this->pid = pid;
+}
+
```
Comment (lychanl) [inline utils/src/OperationWithPathRequest.cpp]: Unnecessary, please remove

File: utils/src/IPCPackets.h [hunk 4]
```diff
@@ -55,6 +55,9 @@ namespace simplefs
 
 		virtual void serialize(char* data);
 
+		int getPid();
+		void setPid(int);
+
```
Comment (lychanl) [inline utils/src/IPCPackets.h]: Unnecessary, please remove

File: daemon/src/communication/ServerListener.cpp (hunk 1 above)
Comment (nkp123) [reply daemon/src/communication/ServerListener.cpp]: Done

File: daemon/src/management/FileDescriptor.h (hunk 2 above)
Comment (nkp123) [reply daemon/src/management/FileDescriptor.h]: No, but there is a wrapper within open function which converts linux flags into these above.

File: utils/src/OperationWithPathRequest.cpp (hunk 3 above)
Comment (nkp123) [reply utils/src/OperationWithPathRequest.cpp]: Done

File: utils/src/IPCPackets.h (hunk 4 above)
Comment (nkp123) [reply utils/src/IPCPackets.h]: Done

File: utils/src/IPCPackets.h [hunk 5]
```diff
@@ -67,6 +67,7 @@ namespace simplefs
 		OperationWithPathRequest(OperationWithPathRequest&&);
 		OperationWithPathRequest& operator=(OperationWithPathRequest);
 
+		int pid;
```
Comment (lychanl) [inline utils/src/IPCPackets.h]: To remove, unused

**Pull request merged**